Developing progress

MihailNenahov opened the pull request:
**Current progress:**

1. wrote migrations
2. functions for jwt
3. html forms
4. package for postgres work
5. some structures(User, Claims, Tweet)


**Some notes:**

- Not sure, that i correctly getting info form user. Right now i can get it from http form, but it makes more difficult routing. 
- Still couldn't try it on database, so it can have some bugs.
- Need to cover it with tests.
- Added some kind of "home page". It wasn't given in task. It's just my wish.

Commit:
made some progress

current progress:
wrote migrations
some functions for jwt
html forms
package for postgres work
some structures

Mishamba commented:
Found out that my code that generates jwt token can be improved.

Commit:
made some progress

rewrote registration
added some code for jwt processing
made migration names valid
changed smth in data_process(i don't remember)))

Commit:
made some progress

now all functions get users info from JWT token
found some bugs

Mishamba commented:
Middle level almost done. Need to make code more stylized and try it on database.

Commit:
made some progress

changed error page
made my code more stylized
changed connection string(to database)
renamed migration files
tring to add subscribe function

Mishamba commented:
Finished middle level. Still have problems with connection to database. 

**I have errors like:**

- github.com/golang-migrate/migrate/v4 is misssing
- if in database i write `CREATE DATABASE tweeter;` everything is ok, but when i write `SHOW DATABASES;` i have error.
- also still not sure how to write subscribe function

Commit:
made some progress

edited html files. now they send post request

Commit:
final version

finished
